packages/devextreme/js/__internal/ui/splitter/splitter.ts

```diff
@@ -659,14 +668,107 @@ class Splitter extends CollectionWidgetLiveUpdate<Properties> {
       case 'collapsible':
         this._updateResizeHandlesCollapsibleState();
         break;
-      case 'visible':
-        this._invalidate();
+      case 'visible': {
+        super._itemOptionChanged(item, property, value, prevValue);
+
+        this._itemVisibleOptionChanged(item, value as boolean, prevValue as boolean);
         break;
+      }
       default:
         super._itemOptionChanged(item, property, value, prevValue);
     }
   }
 
+  _itemVisibleOptionChanged(item: Item, value: boolean, prevValue: boolean): void {
```

Comment:
Check condition is confusing, as long we defined `value` and `prevValue` as booleans, `Item` interface allows them to be undefined.
Can we simplify it? 
If value is undefined, will the later logic work properly?

```suggestion
  _itemVisibleOptionChanged(item: Item, value = true, prevValue = true): void {
      if (prevValue === value) {
```

Reply:
The base logic is not working correctly in cases where the value resets to the default state, which is why we need to add an additional condition. In theory, itemOptionChanged should not be called in some cases—when changing from undefined to true and vice versa -because it's the same meaning in our scenario.

I would like to safely cast to a boolean if something different from the user was passed, as it looks safer. Unfortunately, in jQuery, it can pass values like 0, 1, '', or 'some string'.

Reply:
https://github.com/DevExpress/DevExtreme/pull/28876/commits/58a0820fcd17a0422a9d90b12eaee01c5e286602